src/transaction/sign.ts

```diff
@@ -55,12 +49,61 @@ function signWithKeypair(
   }
 
   const serialized = binary.encode(tx)
+
+  // Decode the serialized transaction:
+  const decoded = binary.decode(serialized)
+
+  // ...And ensure it is equal to the original tx, except:
+  // - It must have a TxnSignature or txSigners (multisign).
```

Comment:
What about flags? The "standard" way of serializing a transaction usually adds tfFullyCanonicalSig even if the user didn't specify it. I could see that feature accidentally tripping this check.

Reply:
That's a great question. I think that if the user didn't specify `tfFullyCanonicalSig` then ripple-lib's `sign` method does not add it. However, the `prepare*` methods -- including `prepareTransaction` -- do: https://github.com/ripple/ripple-lib/blob/develop/src/transaction/utils.ts#L67

Reply:
I thought if you left `Flags` off entirely it _did_ provide it.

Reply:
- I've verified that signing a transaction with ripple-lib's `sign` method (which also serializes it) does not add tfFullyCanonicalSig, regardless of whether the user specified Flags or not. But note that the `prepare*` methods do add it.
- I've added a test that ensures that no Flags are added: https://github.com/ripple/ripple-lib/pull/1026/commits/c5714deddcebd496e9f248e7fee5ab1f697d2320

Reply:
I think that despite the `prepare*` methods, the `sign` method should also add `Flags` if none are specified, mainly because our documentation says we do: https://xrpl.org/transaction-malleability.html

> To further protect users, Ripple has configured its code to enable the `tfFullyCanonicalSig` flag by default where possible. Ripple strongly encourages third-party implementations of XRP Ledger software to generate only fully-canonical signatures, and enable tfFullyCanonicalSig on transactions by default.

Reply:
@ximinez I think that's beyond the scope of this PR, but I've created a new PR that does that: https://github.com/ripple/ripple-lib/pull/1028